demoScripts/quickstart-nodejs.md

```diff
@@ -0,0 +1,529 @@
+
+- Have an Azure account with an active subscription. [Create an account for free](https://azure.microsoft.com/free/?utm_source=campaign&utm_campaign=vscode-tutorial-app-service-extension&mktingSource=vscode-tutorial-app-service-extension).
+- Install [Node.js LTS and npm](https://nodejs.org). Run the command `node --version` to verify that Node.js is installed.
+- Have an FTP client (for example, [FileZilla](https://filezilla-project.org)), to connect to your app.
+
+::: zone-end
+
+## Create your Node.js application
+
+In this step, you create a basic Node.js application and ensure it runs on your computer.
+
+> [!TIP]
+> If you have already completed the [Node.js tutorial](https://code.visualstudio.com/docs/nodejs/nodejs-tutorial), you can skip ahead to [Deploy to Azure](#deploy-to-azure).
+
+1. Create a Node.js application using the [Express Generator](https://expressjs.com/starter/generator.html), which is installed by default with Node.js and NPM.
+
+    ```bash
+    npx --yes express-generator $EXPRESS_APP_NAME --view ejs 
+    ```
+
+1. Change to the application's directory and install the NPM packages.
+
+    ```bash
+    cd $EXPRESS_APP_NAME && npm install
+    ```
+
+1. If you are running through this manually, run `DEBUG=myexpressapp:* npm start` and open a browser and navigate to `http://localhost:3000`. You should see something like this:
+
+    ![Running Express Application](./media/quickstart-nodejs/express.png)
+
+* If you would prefer to check this in an automated manner run the following script:
+
+    ```bash
+    npm start&
+    RUNTIME="30 seconds"
+    END_TIME=$(date -ud "$RUNTIME" +%s)
+    while [[ $(date -u +%s) -le $END_TIME ]]; \
+    do RESPONSE=$(curl -sL -w "%{http_code}" localhost:3000 -o /dev/null); \
+    if [ "$RESPONSE" == "200" ]; then break; \
+    else sleep 5; \
+    fi; \
+    done
+    echo $RESPONSE
+    PID=$(lsof -t -i tcp:3000) 
+    kill $PID
+    ```
+    <!--expected_similarity=0.5-->
+    ```Output
+    [1] 11339
+
+    > myexpressapp@0.0.0 start
+    > node ./bin/www
+
+    GET / 200 5.991 ms - 207
+    200
+    ```
+:::zone target="docs" pivot="development-environment-vscode"
+> [!div class="nextstepaction"]
+> [I ran into an issue](https://www.research.net/r/PWZWZ52?tutorial=node-deployment-azure-app-service&step=create-app)
+::: zone-end
+## Deploy to Azure
+
+Before you continue, ensure that you have all the prerequisites installed and configured.
+
+> [!NOTE]
+> For your Node.js application to run in Azure, it needs to listen on the port provided by the `PORT` environment variable. In your generated Express app, this environment variable is already used in the startup script *bin/www* (search for `process.env.PORT`).
+>
+
+:::zone target="docs" pivot="development-environment-vscode"
+
+#### Sign in to Azure
+
+1. In the terminal, ensure you're in the *myExpressApp* directory, then start Visual Studio Code with the following command:
+
+    ```bash
+    code .
```

Comment:
Won't this be executed by Innovation Engine? That's bad, right? 

I imagine this will just silently fail since there is no results block so it won't be an immediate problem in the testing use case, but it would be in the learning use case. Do we need to raise an issue to parse the "pivot=" and ignore vs code areas? Or perhaps this is a special command that needs to be ignored?

Reply:
This silently fails. Well, it actually opens VSCode if it is installed haha... but yes. We need to figure out some way to handle these types of commands. How do we do so without breaking the rule we set out. Or do we need to change the rule?

Rule for context: "It breaks the rule that we should not have a different behavior in the engine vs standalone execution. This is a very important rule and should always be enforced. Failure to do so means that we are no longer testing the thing that is going into the customers hands"